src/main/java/edu/ithaca/dturnbull/bank/Account/AbstractAccount.java

```diff
@@ -0,0 +1,85 @@
+package edu.ithaca.dturnbull.bank.Account;
+
+import java.util.List;
+
+public abstract class AbstractAccount {
+    String email;
+    double balance;
+    List<String> history;
+
+    /**
+    * @post @return balance
+    */
+    public double getBalance(){
+        return balance;
+    }
+    
+    /**
+    * @post @return email
+    */
+    public String getEmail(){
+        return email;
+    }
+
+    /**
+    * @post increases the balance by @param amount if amount is non-negative
+    */
+    public void deposit(double amount) throws IllegalArgumentException{
+        if (isNumberValid(amount)) {
+            balance += amount;
+            balance = Math.round(balance * 100.0) / 100.0;
+            appendTransaction(amount, "deposit");
+        } else {
+            throw new IllegalArgumentException("Amount to deposit is invalid");
+        }
+    }
+
+    /**
+     * @post reduces the balance by @param amount if amount is non-negative and smaller than balance
+     */
+    public abstract void withdraw(double Amount)throws InsufficientFundsException;
+
+    /**
+     * @post reduces the balance by @param amount if amount is non-negative and smaller than balance
+     * increases the balance of @param transferee 
+     */
+    public abstract void transfer(double amount, AbstractAccount transferee)  throws InsufficientFundsException;
+
+    public String historyToString(){
+        String temp = "";
+        for (int i = 0; i < history.size(); i++){
+            temp+=history.get(i);
+        }
+        return temp;
+    }
+
+    /**
+     * @post checks to see if the @param email is valid
+     */
+    public void appendTransaction(double amount, String action){
+        this.history.add(toString(amount, action));
+    }
+    
+    public String toString(double amount, String action){
```

Comment:
Be careful using a methon named "toString". This method name is usually reserved for overriding when printing the object itself.